llvm-external-projects/iree-dialects/lib/Dialect/VectorExt/IR/VectorExtAttrs.cpp

```diff
@@ -324,11 +420,20 @@ NestedLayoutAttr::computeThreadIds(Value threadId,
   auto tileSizes = llvm::concat<const int64_t>(
       applyPermutation(getSubgroupsPerWorkgroup(), getSubgroupOrder()),
       applyPermutation(getThreadsPerOuter(), getThreadOrder()));
+  auto tileSizesIterator = tileSizes.begin();
+
+  auto activeIdFilter =
+      llvm::concat<const bool>(getSubgroupActiveIds(), getThreadActiveIds());
 
   // Modulo the delinearized subgroup/thread ids by the number of unique
   // elements distributed to those ids.
-  for (auto [delinearized, basis, tile] :
-       llvm::zip(delinearized, basisSizes, tileSizes)) {
+  for (auto [delinearized, basis, isActive] :
```

Comment:
Can we potentially put more comments here? It's very trickly to follow the logic here so any additional document could help a lot for maintenance.

Reply:
I added a bunch :)

Hopefully that helps, although it is still a lot of moving pieces...